crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -569,6 +604,205 @@ pub fn if_with_same_arms(checker: &mut Checker, stmt: &Stmt, parent: Option<&Stm
     }
 }
 
+fn should_proceed_main(test: &Expr, body: &[Stmt], orelse: &[Stmt]) -> bool {
+    if let ExprKind::Compare {
+        left,
+        ops,
+        comparators,
+    } = &test.node
+    {
+        if let ExprKind::Name { .. } = &left.node {
+            if ops.len() == 1 && ops[0] == Cmpop::Eq {
+                if comparators.len() == 1 {
+                    if let ExprKind::Constant { .. } = &comparators[0].node {
+                        if body.len() == 1 {
+                            if let StmtKind::Return { .. } = &body[0].node {
+                                if orelse.len() == 1 {
+                                    if let StmtKind::If { .. } = &orelse[0].node {
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+fn should_proceed_child(stmt: &Stmt, var_id: &str) -> bool {
+    if let StmtKind::If { test, body, orelse } = &stmt.node {
+        if let ExprKind::Compare {
+            left,
+            ops,
+            comparators,
+        } = &test.node
+        {
+            if let ExprKind::Name { id, .. } = &left.node {
+                if id == var_id && ops.len() == 1 && ops[0] == Cmpop::Eq {
+                    if comparators.len() == 1 {
+                        if let ExprKind::Constant { .. } = &comparators[0].node {
+                            if body.len() == 1 {
+                                if let StmtKind::Return { .. } = &body[0].node {
+                                    if orelse.len() <= 1 {
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+fn constant_to_str(value: &Constant) -> String {
+    match value {
+        Constant::None => "None".to_string(),
+        Constant::Str(s) => format!("{:?}", s),
+        Constant::Int(i) => i.to_string(),
+        Constant::Float(f) => f.to_string(),
+        Constant::Ellipsis => "...".to_string(),
+        Constant::Bytes(b) => {
+            let the_string = String::from_utf8(b.clone()).unwrap();
```

Comment:
What happens if the byte sequence isn't valid utf-8? I would expect this call to fail. Would it be better to change the return type to `Option<String>` or `Result<SomeError, String>` so that the method can return `None` if the constant cannot be converted to a string?